lib/schema/boolean.js

```diff
@@ -224,6 +224,8 @@ SchemaBoolean.prototype.castForQuery = function($conditional, val) {
   return this._castForQuery($conditional);
 };
 
+SchemaBoolean.prototype.castNull = true;
```

Comment:
Is this change necessary given your other changes? Your code example doesn't make any reference to this new `castNull` option.

Reply:
It's being used in [this line](https://github.com/Automattic/mongoose/blob/18296bb5d6fdc9975e2825dd4fee995fe340c516/lib/schematype.js#L1058).

Reply:
I just used it because it's more extensible than say type checking for Boolean in the line @AbdelrahmanHafez referred to. If in future, other types want to cast null, they'd set `castNull = true` to get passed that conditional check, although currently other type cast implementations (like ex. [Number](https://github.com/Automattic/mongoose/blob/43288f869366ade4c287b4dd06e9c7016c6104c1/lib/cast/number.js#L17)) don't do anything with null even if castNull is true. Boolean is a bit of a special case because there is an existing documented mechanism (convertToTrue|False) exposed to easily customize null casting.

Reply:
If this is approved, it would be great if you add this option to the documentation using JSDoc.